builtin/providers/aws/resource_aws_ecr_repository.go

```diff
@@ -81,9 +87,18 @@ func resourceAwsEcrRepositoryRead(d *schema.ResourceData, meta interface{}) erro
 	d.Set("arn", *repository.RepositoryArn)
 	d.Set("registry_id", *repository.RegistryId)
 
+	repositoryUrl := buildRepositoryUrl(repository, meta)
+	log.Printf("[INFO] Setting the repository url to be %s", repositoryUrl)
+	d.Set("repository_url", repositoryUrl)
+
 	return nil
 }
 
+func buildRepositoryUrl(repo *ecr.Repository, meta interface{}) string {
```

Comment:
It would be nice to make this function less greedy in terms of arguments. e.g.

```go
func buildRepositoryUrl(repo *ecr.Repository, region string) string {
```